gdnative-core/src/geom/basis.rs

```diff
@@ -29,7 +33,70 @@ impl Basis {
         }
     }
 
-    pub fn from_diagonal(p_diag: Vector3) -> Basis {
+    fn invert(&mut self) {
+        let co = vec![
+            self.elements[1].y * self.elements[2].z - self.elements[1].z * self.elements[2].y,
+            self.elements[1].z * self.elements[2].x - self.elements[1].x * self.elements[2].z,
+            self.elements[1].x * self.elements[2].y - self.elements[1].y * self.elements[2].x,
+        ];
+
+        let det =
+            self.elements[0].x * co[0] + self.elements[0].y * co[1] + self.elements[0].z * co[2];
+
+        debug_assert_ne!(det, 0.0, "Determinant was zero");
```

Comment:
The C++ code returns early leaving the object in an unexpected state if MATH_CHECKS is defined. The asserts were to remind me what the checks were, but I'm not sure if we want to mimic that in the Rust code; ie: define cfg var and bail early, etc. What is thy preference?

Reply:
Well... You just use a regular `assert`? It will panic if the condition failed and unwind the stack. I don't know what's the problem here. The original issue is about API *availability*, not copying the exact behavior 1:1. Yes, it is required that sanity tests are added, but that's for *correctness*. There is a difference between correctness and bug-compatibility.